### PR TITLE
video details added

### PR DESCRIPTION
**Pull request by daiyankhan**

### CODE REVIEW

**Comment by thehungrycoder** [comment]
This looks good, thanks @daiyankhan. I was envisioning this slightly differently. 

those two existing methods will just return ids. we can create a new public method to get details. we can pass one or more video ids to that function and it'll get the details. something like below

```
function getVideoDetails(videoIds = []) {
  videoIds.forEach((videoId) => {})
```

So, in one request, we'll get just video ids. and in another request we'll get details for those videos. the reason i'm thinking, we don't need all video's details all times. once we got info about a video, we'll cache it. next time we will just get list and see if we already have info in our db. if we don't have info in our db, we'll get the details. let me know if it doesn't make sense.

**Comment by daiyankhan** [comment]
Okay I will change accordingly. Thank you for elaborate description.

File: scrap.js [hunk 1]
```diff
@@ -61,15 +61,18 @@ const getChannelVideos = async (channelId, maxPage)  => {
...
-  const videoDetail =  await youtube.searchOne(url)
-  console.log('video detail', videoDetail)
-
-  return videoDetails;
+  return videos;
 }
-)}
 
+const getVideoDetails = (videos = []) => {
+  videos.map( async (videoId) => {
```
Comment (thehungrycoder) [inline scrap.js:68]: you need to return the map!

Comment (thehungrycoder) [inline scrap.js:68]:
```
return videos.map( async (videoId) => { 
```

File: scrap.js [hunk 2]
```diff
@@ -81,10 +84,12 @@ const getUserVideos = async (userId, maxPage)  => {
   return videos;
 }
 
-getChannelVideos('UCCMC_4hcI9zoOvjSfka0-xQ', 5);
+// getChannelVideos('UCCMC_4hcI9zoOvjSfka0-xQ', 5);
 // getUserVideos('ValleyRanchIslamicCenter', 5);
+getVideoDetails(['qSP_BP-pM4w', 'ra-5KkKrbXg', '-7DHxq4PBoQ', 'zZgcOpJOAWo'])
```
Comment (thehungrycoder) [inline scrap.js:89]: let's comment this. it gonna otherwise run for all request!

File: scrap.js (hunk 1 above)
Comment (daiyankhan) [reply scrap.js:68]: sorry about that, changing it now

File: scrap.js (hunk 2 above)
Comment (daiyankhan) [reply scrap.js:89]: done

**Pull request closed without merging**